cmd/snap-confine/snap-confine.c

```diff
@@ -123,6 +123,22 @@ int main(int argc, char **argv)
 			debug
 			    ("skipping sandbox setup, classic confinement in use");
 		} else {
+			/* snap-confine uses privately-shared /run/snapd/ns to store
+			 * bind-mounted mount namespaces of each snap. In the case that
+			 * snap-confine is invoked from the mount namespace it typically
+			 * constructs, the said directory does not contain mount entries
+			 * for preserved namespaces as those are only visible in the main,
+			 * outer namespace.
+			 *
+			 * In order to operate in such an environment snap-confine must
+			 * first re-associate its own process with another namespace in
+			 * which the /run/snapd/ns directory is visible.  The most obvious
+			 * candidate is pid one, which definitely doesn't run in a
+			 * snap-specific namespace, has a predictable PID and is long
+			 * lived.
+			 */
+			sc_reassociate_with_pid1_mount_ns();
+			const char *snap_name = getenv("SNAP_NAME");
```

Comment:
We don't need this. I think it was a part of a very very old fix that can now be simplified